### PR TITLE
Add instrumentation helpers: event_loggers and load_state_stats

### PR DESCRIPTION
**Pull request by dantswain**

### CODE REVIEW

File: lib/pay_day_loan.ex [hunk 1]
```diff
@@ -296,6 +316,25 @@ defmodule PayDayLoan do
...
 
   @doc """
+  Returns a map of load states and the number of keys in each state
+
+  Useful for instrumentation
+  """
+  @spec load_state_stats(pdl :: t) :: %{}
+  def load_state_stats(pdl = %PayDayLoan{}) do
+    stats = %{requested: 0, loaded: 0, loading: 0, failed: 0}
+    # this is stepping over the boundary of the PDL - this function could
```
Comment (joshuawscott) [inline lib/pay_day_loan.ex]: Is this a leftover comment?

Comment (dantswain) [reply lib/pay_day_loan.ex]: Woops, yes.

File: test/pay_day_loan/backends/generic_test.exs [hunk 2]
```diff
@@ -251,6 +287,8 @@ defmodule PayDayLoan.Backends.GenericTest do
     assert nil == PDL.LoadState.peek(Cache.pdl().load_state_manager, key)
     # we hold onto the knowledge that the key exists
     assert PDL.KeyCache.in_cache?(Cache.pdl().key_cache, key)
+
+    assert [disappeared: key]
```
Comment (joshuawscott) [inline test/pay_day_loan/backends/generic_test.exs]: `== CacheLogger.logs`

File: test/pay_day_loan/backends/generic_test.exs [hunk 3]
```diff
@@ -182,6 +214,8 @@ defmodule PayDayLoan.Backends.GenericTest do
     assert {:error, :failed} == Cache.get(key)
     # should get cleared from the load state cache
     assert nil == PDL.peek_load_state(Cache.pdl(), key)
+
+    assert [failed: key]
```
Comment (joshuawscott) [inline test/pay_day_loan/backends/generic_test.exs]: Also here

**Review by joshuawscott — APPROVED** [review]
:shipit: after comments

**Review by CoitThomas — APPROVED** [review]
++ to Josh's comments, :shipit:

**Pull request merged**